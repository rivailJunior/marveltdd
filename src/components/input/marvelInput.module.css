.showClear {
	display: block;
}

.hideClear {
	display: none !important;
}

.inputSearch {
	background: #ffffff 0% 0% no-repeat padding-box;
	border: 1px solid #e5e5e5;
	border-radius: 4px;
	opacity: 1;
	text-align: left;
	font: italic normal normal 14px/19px PT Sans;
	letter-spacing: 0px;
	color: #8e8e8e;
	width: 295px;
	height: 32px !important;
	border-radius: 3px !important;
}

.labelSearch {
	margin-top: 5%;
	text-align: left;
	font: normal normal bold 16px/24px PT Sans Caption;
	letter-spacing: 0px;
	color: #555555;
	opacity: 1;
}

.icon {
	max-width: 15px;
	position: absolute;
	right: 1;
	right: 10px;
	justify-self: center;
	align-self: center;
}

.inputGroup {
	justify-content: center;
	display: flex;
	align-items: center;
}
