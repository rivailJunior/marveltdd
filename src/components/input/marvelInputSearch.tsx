import { useState, useEffect } from "react";
import { InputGroup, FormControl } from "react-bootstrap";
import styles from "./marvelInput.module.css";

type iInputProps = {
    handleChangeName: Function;
};
const MarvelInputSearch = ({ handleChangeName }: iInputProps) => {
    const [name, setName] = useState("");
    useEffect(() => {
        if (name.length > 1) {
            handleChangeName(name);
        }
    }, [name]);

    const handleClear = () => {
        setName("");
        handleChangeName("");
    };
    const showOrHideClear =
        name.length > 0 ? styles.showClear : styles.hideClear;
    return (
        <InputGroup className="mb-3">
            <FormControl
                value={name}
                onChange={(evt) => setName(evt.target.value)}
                data-testid="inputSearch"
                placeholder="Search"
                aria-label="Busque personagens"
            />
            <InputGroup.Append data-testid="btnSearch">
                <InputGroup.Text>OK</InputGroup.Text>
            </InputGroup.Append>
            <InputGroup.Append
                className={showOrHideClear}
                data-testid="btnClear"
                onClick={handleClear}
            >
                <InputGroup.Text>X</InputGroup.Text>
            </InputGroup.Append>
        </InputGroup>
    );
};

export { MarvelInputSearch };
