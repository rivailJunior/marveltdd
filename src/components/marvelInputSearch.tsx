import { useState, useEffect } from "react";
import { InputGroup, FormControl } from "react-bootstrap";
import styles from "./marvelInput.module.css";

type iInputProps = {
    handleChangeName: Function;
};
const MarvelInputSearch = ({ handleChangeName }: iInputProps) => {
    const [name, setName] = useState("");
    useEffect(() => {
        handleChangeName(name);
    }, [name, handleChangeName]);
    const showOrHideClear =
        name.length > 0 ? styles.showClear : styles.hideClear;
    return (
        <InputGroup className="mb-3">
            <FormControl
                value={name}
                onChange={(evt) => setName(evt.target.value)}
                data-testid="inputSearch"
                placeholder="Search"
                aria-label="Busque personagens"
            />
            <InputGroup.Append data-testid="btnSearch">
                <InputGroup.Text>OK</InputGroup.Text>
            </InputGroup.Append>
            <InputGroup.Append
                className={showOrHideClear}
                data-testid="btnClear"
                onClick={() => setName("")}
            >
                <InputGroup.Text>X</InputGroup.Text>
            </InputGroup.Append>
        </InputGroup>
    );
};

export { MarvelInputSearch };
