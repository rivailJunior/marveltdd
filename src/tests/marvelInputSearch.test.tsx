import { render, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MarvelInputSearch } from "../components/input/marvelInputSearch";

afterEach(cleanup);

describe("`<MarvelInputSearch>`", () => {
    test("Render correctly", () => {
        const container = renderer.create(
            <MarvelInputSearch handleChangeName={() => { }} />
        );
        container.toJSON();
        expect(container).toMatchSnapshot();
    });

    test("Show clear button when type", async () => {
        const { getByTestId } = render(
            <MarvelInputSearch handleChangeName={() => { }} />
        );
        const input = getByTestId("inputSearch");
        fireEvent.change(input, {
            target: {
                value: "Capitao america",
            },
        });
        const clearButton = getByTestId("btnClear");
        expect(clearButton).toHaveClass("showClear");
    });

    test("Clear input when click at clear button and hide clear button", () => {
        const { getByTestId } = render(
            <MarvelInputSearch handleChangeName={() => { }} />
        );
        const input = getByTestId("inputSearch");
        fireEvent.change(input, {
            target: {
                value: "Capitao america",
            },
        });
        const clearButton = getByTestId("btnClear");
        expect(clearButton).toHaveClass("showClear");
        fireEvent.click(clearButton);
        expect(input.value).toBe("");
        expect(clearButton).toHaveClass("hideClear");
    });

    test("Fire callback to get current input name typed", () => {
        const getName = jest.fn();
        const { getByTestId } = render(
            <MarvelInputSearch handleChangeName={getName} />
        );
        const input = getByTestId("inputSearch");
        fireEvent.change(input, {
            target: {
                value: "Capitao america",
            },
        });
        const searchBtn = getByTestId("btnSearch");
        fireEvent.click(searchBtn);
        expect(getName).toBeCalledTimes(1);
        expect(getName).toBeCalledWith("Capitao america");
    });
});
