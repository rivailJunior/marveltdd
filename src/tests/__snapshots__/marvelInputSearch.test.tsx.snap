// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`<MarvelInputSearch>\` Render correctly 1`] = `
<div
  className="mb-3 input-group"
>
  <input
    aria-label="Busque personagens"
    className="form-control"
    data-testid="inputSearch"
    onChange={[Function]}
    placeholder="Search"
    value=""
  />
  <div
    className="input-group-append"
    data-testid="btnSearch"
  >
    <span
      className="input-group-text"
    >
      OK
    </span>
  </div>
  <div
    className="hideClear input-group-append"
    data-testid="btnClear"
    onClick={[Function]}
  >
    <span
      className="input-group-text"
    >
      X
    </span>
  </div>
</div>
`;
